server/service/apple_mdm.go

```diff
@@ -341,15 +348,28 @@ func (svc *Service) NewMDMAppleConfigProfile(ctx context.Context, teamID uint, r
 
 	newCP, err := svc.ds.NewMDMAppleConfigProfile(ctx, *cp)
 	if err != nil {
+		var existsErr existsErrorInterface
+		if errors.As(err, &existsErr) {
+			err = fleet.NewInvalidArgumentError("profile", "Couldn't upload. A configuration profile with this name already exists.").
+				WithStatus(http.StatusConflict)
```

Comment:
Standardized the error message to be the same as for Windows. It could be argued that this is a breaking change for when this is called from the old endpoint, but the details of error responses are not documented, IMO this is fine.